### PR TITLE
8195675: Call to insertText with single character from custom Input Method ignored

### PR DESCRIPTION
**Pull request by srl295**
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8195675](https://bugs.openjdk.org/browse/JDK-8195675): Call to insertText with single character from custom Input Method ignored (**Bug** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17921/head:pull/17921` \
`$ git checkout pull/17921`

Update a local copy of the PR: \
`$ git checkout pull/17921` \
`$ git pull https://git.openjdk.org/jdk.git pull/17921/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17921`

View PR using the GUI difftool: \
`$ git pr show -t 17921`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17921.diff">https://git.openjdk.org/jdk/pull/17921.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17921#issuecomment-1972580375)

### CODE REVIEW

File: src/java.desktop/macosx/native/libawt_lwawt/awt/AWTView.m [hunk 1]
```diff
@@ -965,7 +968,7 @@ - (void) insertText:(id)aString replacementRange:(NSRange)replacementRange
 
     if ((utf16Length > 2) ||
         ((utf8Length > 1) && [self isCodePointInUnicodeBlockNeedingIMEvent:codePoint]) ||
-        ((codePoint == 0x5c) && ([(NSString *)kbdLayout containsString:@"Kotoeri"]))) {
+        [(NSString *)kbdLayout containsString:@KEYMAN_LAYOUT]) {
 #ifdef IM_DEBUG
```
Comment (prrace) [inline src/java.desktop/macosx/native/libawt_lwawt/awt/AWTView.m:972]: As noted in JBS Kotoeri is obsolete, which is why the reference to it here should be removed as part of this change.

**Review by prrace — APPROVED** [review]

**Comment by srl295** [comment]
I'm able to build this again, but not satisfied that it solves the issue, so I am still investigating. 
Should I rebase this upon latest master or somewhere else (now that 22 is released)? @prrace

**Comment by prrace** [comment]
> I'm able to build this again, but not satisfied that it solves the issue, so I am still investigating. Should I rebase this upon latest master or somewhere else (now that 22 is released)? @prrace

@srl295 
You are based on main-line, which is now 23, so there should be nothing to do.
If you feel you are out of date in some important way you can re-base but the release change would have nothing to do with that.

**Comment by srl295** [comment]
Great. I'll keep investigating

**Comment by srl295** [comment]
Just got my Mac setup again today - will commence testing

**Comment by srl295** [comment]
@prrace i want to verify it fixes the issue.

**Comment by srl295** [comment]
Great news, this fixes the issue!!

image

**Comment by srl295** [comment]
@prrace @naotoj recommend merging this once it passes appropriate tests.

**Comment by srl295** [comment]
@prrace any hope of backporting this? I can do it if you point me at the steps

**Review by prrace — APPROVED** [review]
@srl295 
You just need to add a comment saying
/integrate
I can't do that for you.

Since you are a JDK committer, it will then be pushed

**Comment by prrace** [comment]
> @prrace any hope of backporting this? I can do it if you point me at the steps

@srl - that's not my call. The update release maintainers are in charge

The general process is https://openjdk.org/projects/jdk-updates/approval.html
and the list of update releases and any specific policies they may have are here
https://openjdk.org/projects/jdk-updates/

**Comment by srl295** [comment]
> @srl295 You just need to add a comment saying /integrate I can't do that for you.
> 
> Since you are a JDK committer, it will then be pushed

Thanks. I think I might not try it on the weekend before a US holiday :)  But i'll do it Tuesday

**Comment by srl295** [comment]
@prrace hi- opened a backport as https://github.com/openjdk/jdk21u-dev/pull/1052 with a backport JBS

**Comment by srl295** [comment]
@prrace the backports seem to be progressing well.. there's a report that it's not working with JavaFX, though.  I'm not familiar with JFX enough to know about this.  Do you think there's either a special backport needed, or could there be a *parallel* fix in JFX that's needed? 

@AndyBlack created https://github.com/AndyBlack/KeymanFX and https://github.com/AndyBlack/KeymanSwing as test cases. It seems that on the JFX side it's using 
 `javafx.scene.control.TextField` - should that be affected (that is, fixed) by this fix to AWTView or would JFX be perhaps similar parallel code needing a parallel fix?

**Pull request closed without merging**